### ISSUE TITLE
Consider splitting the /data/ into it's own repository.

### ISSUE THREAD

**Issue by augustjohnson** [open5e/open5e-api#148]
I really like the concept of separation of concerns, so that the API is pure python, but at build time it pulls data (which is nicely json formatted) into itself. Then the data can be freely updated without python being touched, and a nice landing page can be made that includes the data.
I would suggest even making jekyll pages and possibly even a smaller, static-api style json page. It could not include search, but it could include plenty of other stuff.
https://jekyllrb.com/docs/datafiles/
Basically:
open5e-data would primarily be a json data repo, and would have a couple of front-matter type pages in jekyll that would allow for simple browsing of the site. It could have a nice readme, and all content could be properly licensed as CC.

**Impacts:**
New repo, including some new documentation.
New API build process, where the API would pull from a live repo. Maybe the open5e-data repo has build artifacts?
Some updates to the frontend to create those references.

**Comment by augustjohnson** [comment]
As a note, images could follow this model, either going into the open5e-data repo (maybe renamed to open5e-assets to be data-type-agnostic) or possibly a distinct open5e-images repo.

**Comment by augustjohnson** [comment]
To flesh this out a bit more, I'm going to bring up a few use cases, and try and flesh out how a distinct data repo would be useful for them.

**1. A user of open5e.com sees a data error.**
In this scenario, a simple call to action link on each data display page could say something like "See a data error? Log a bug here!" with a link to the open5e-data repo issues with well formated query parameters that offer a simple reference. This offers a simple way to crowdsource error data into a self-contained repository.

**2. A mobile app developer wants to use JUST our well-formatted json, they have no need for outbound calls to our API.**
In this scenario, a highly-automated mobile dev would simply want to have a build process that made an outbound GET to a well-formatted data source (maybe a .zip of the json). This could be delivered by the open5e-data repo's RELEASES feature, which would have tags that are pinnable like "latest" and "2023.04" for an April 2023 release, or something along those lines. These versions could be hard-referenced downstream as well, for example, a foundry plugin could pin themselves to 2023.04 until they manually tested new monsters added in 2023.07.

**3. A self-hoster of our API wants to add custom magic spells for their own world.**
In our populate-db script to input data to our API, we would (by default) make an outbound network request to open5e-data RELEASE :latest. It would pull down and unzip, and then import the json just like before. This outbound call could be overridden with a local .zip reference, or a call to a different URL, allowing for the local hoster to add their own spells, even from their own forked github repo of our data repo. This would require almost 0 python skills.

**4. Someone who finds json too scary wants to use well-formatted DND data.**
We could create a **release** of our JSON data that goes through a simple converter to CSV. This would be available under the RELEASES section of github.

I truly think that the data itself is one of our key products, as opposed to the API. It's so small, and properly licensed to be shared and spread around all over. This treats the data as it's own product, and would allow for many downstream consumers, (the API being one).

**Comment by dprothero** [comment]
Using releases makes a ton of sense here. You can even make the creation of a release be the trigger for a GitHub action to deploy the web site, send notifications of the new version, etc.

**Comment by dprothero** [comment]
And to pile on, this is a great idea with tons of upside.

**Comment by augustjohnson** [comment]
@dprothero gave me this link as a place to start if I'm building custom json schemas.
https://www.liquid-technologies.com/online-json-to-schema-converter
I think the idea would be that any changes to any of the json files would go through a schema check.
https://github.com/python-jsonschema/check-jsonschema

**Comment by dprothero** [comment]
Greg makes some good points here we should consider. I do think having them separate removes the complexity for content-only contributors (e.g. why is my PR for a new spell failing on some Python code?)

Either way, I like the idea of doing the JSON schema thing to validate PRs to the data.

**Comment by augustjohnson** [comment]
Yeah that's a great perspective.

So if we want to use the "releases" feature of github to just produce a well-documented DATA ONLY package, that would still be very possible, within the current repo.

How about we transition this idea into trying to resolve these rough requirements?

1. Make JUST THE DATA (in static files) very reliably available. (probably using github releases)
2. Smooth the process for one-off typo-fixing.
3. Longer-term, applying schema to the data upon PR.
4. Longer-term making JUST THE DATA available in multiple formats using some sort of automated output script (probably building on the releases features).

**Comment by Gredelston** [comment]
@augustjohnson Sounds great to me. That will let us meet our (users') needs in a much lighter way.

If we decide we want to pursue the separation of concerns, it's definitely doable. Just wanted to point out some hazards.

**Comment by augustjohnson** [comment]
Ok i'm closing this idea out. Negatory.
